src/client/dfs/dfs.c

```diff
@@ -3790,11 +3790,8 @@ readdir_int(dfs_t *dfs, dfs_obj_t *obj, daos_anchor_t *anchor, uint32_t *nr,
 			D_GOTO(out, rc = daos_der2errno(rc));
 
 		for (ptr = enum_buf, i = 0; i < number; i++) {
-			int len;
-
-			len = snprintf(dirs[key_nr].d_name,
-				       kds[i].kd_key_len + 1, "%s", ptr);
-			D_ASSERT(len >= kds[i].kd_key_len);
+			strncpy(dirs[key_nr].d_name, ptr, kds[i].kd_key_len);
+			dirs[key_nr].d_name[kds[i].kd_key_len] = '\0';
```

Comment:
I thought I'd been through daos and remove most of the places where snprintf() was used with "%s" but I must have missed this one.  I think here though as dfs isn't ever doing anything with the entry names then it could just treat them as a memory region and use memcpy().  strncpy() will do two extra steps, check the length of the input string and pad the output buffer with \0, neither of which are of benefit in this case.

The same fix should also be applied to the dfs_listxattr code I assume?

Reply:
agree, memcpy is better in this case.